temba/flows/models.py

```
@@ -2590,6 +2801,337 @@ def clear_cached_child(self):
         if 'cached_child' in self.__dict__:
             del self.__dict__['cached_child']
 
+    @classmethod
+    def create_or_update_from_goflow(cls, session, contact, run_output, run_log, wait, msg_in):
+        """
+        Creates or updates a flow run from the given output returned from goflow
+        """
+        uuid = run_output['uuid']
+        results = run_output.get('results', {})
+        is_active = run_output['status'] in ('active', 'waiting')
+        is_error = run_output['status'] == 'errored'
+        created_on = iso8601.parse_date(run_output['created_on'])
+        exited_on = iso8601.parse_date(run_output['exited_on']) if run_output['exited_on'] else None
```

Maybe we should have a utility method to deal with the parsing of a potentially empty iso8601?

---

Maybe tho arguably there's utility here where in the line above we don't silently carry on if `run_output['created_on']` is null but instead fail fast. Either way I'd prefer to add that in a different pr so we update everywhere we parse ISO dates